pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/store/SegmentDirectoryPaths.java

```diff
@@ -80,7 +80,12 @@ public static File findCreationMetaFile(File indexDir) {
   @Nullable
   public static File findTextIndexIndexFile(File indexDir, String column) {
     String luceneIndexDirectory = column + V1Constants.Indexes.LUCENE_TEXT_INDEX_FILE_EXTENSION;
-    return findFormatFile(indexDir, luceneIndexDirectory);
+    File indexFormatFile = findFormatFile(indexDir, luceneIndexDirectory);
```

Comment:
We should prioritize v9 over v8. Same for other places

Reply:
done.